src/functional/otf/binding/type_specifications.py

```diff
@@ -0,0 +1,27 @@
+from dataclasses import dataclass
+
+import functional.type_system.type_specifications as ts
+from functional import common as func_common
+from functional.type_system.type_specifications import (  # noqa: F401
+    CallableType as CallableType,
+    DataType as DataType,
+    DeferredType as DeferredType,
+    DimensionType as DimensionType,
+    FieldType as FieldType,
+    FunctionType as FunctionType,
+    OffsetType as OffsetType,
+    ScalarKind as ScalarKind,
+    ScalarType as ScalarType,
+    TupleType as TupleType,
+    TypeSpec as TypeSpec,
+    VoidType as VoidType,
+)
```

Comment:
I also agree with @DropD here. I think it would be better to have multiple imports to bring new types in scope instead of trying to expand the original `type_specifications` module (which makes basically impossible to compose type extensions from different modules):
```
# This allows adding extensions from different modules very easily
import functional.type_system.type_specifications as ts
import functional.otf.binding.type_specifications as bts
```